saturn_client/core.py

```diff
@@ -305,6 +306,22 @@ def get_jupyter_server(self, jupyter_server_id) -> Dict[str, Any]:
             ) from err
         return response.json()
 
+    def wait_for_jupyter_server(
+        self, jupyter_server_id, status="running", max_retries=30, sleep_interval=5
```

Comment:
I don't think we should expose parameters about the retry strategy. We should just give people a `timeout` and say "this will block until either the Jupyter is up or more than `timeout` seconds have passed".

Benefits of this:

* we're free to change the retry strategy in the future (say, to exponential backoff to be kinder to atlas) without breaking any customer code
* the method is written in the language that applications are likely to care about. Since Atlas does not have any rate limiting, I think the only thing people will care about is the worst-case time this spends waiting, not the exact number of requests.

Like this:

```python
start_time = datetime.utcnow()
while (datetime.utcnow() - start_time).total_seconds < timeout:
    ....
    if result["status"] == "running":
        break
    else:
        sleep(sleep_interval)
```

I also think we should not expose status strings and just make this method wait until the Jupyter is started. I have a strong preference for keeping the API of this client as small and strict as possible, especially since it doesn't have any tests and hasn't been used by anyone outside of Saturn (so it hasn't received feedback that might change it). The stricter and tighter we make this, the better chance we have that future changes can be made in a non-breaking way.

Reply:
Yeah I like those changes. I was thinking people might want to wait for stop as well, but I agree that case doesn't matter as much.